service/history/queues/queue_base.go

```diff
@@ -339,61 +339,75 @@ func (p *queueBase) checkpoint() {
 		}
 	}()
 
-	totalSlices := 0
 	readerScopes := make(map[int32][]Scope)
-	newInclusiveLowWatermark := tasks.MaximumKey
+	newInclusiveLowWatermark := p.nonReadableScope.Range.InclusiveMin
```

Comment:
should this be exclusiveHighWatermark? since it is the InclusiveMin of non-readablescope, so it should not be readable.

Reply:
Yeah, existing naming is confusing. I renamed it to `exclusiveDeletionHighWatermark`, it should be more clear.